.github/workflows/labeler.yml

```diff
@@ -0,0 +1,15 @@
+# https://github.com/actions/labeler
+name: "Pull Request Labeler"
+on:
+- pull_request_target
```

Comment:
Sorry I'm late to the review party.  @edsantiago WDYT about limiting this trigger to PRs on `main` only?

Background:  The source for GHA workflows is always the default branch - unlike Cirrus-CI.  So over time, with new or existing release-branches, the workflow and other details (like `.github/labeler.yml` contents) may diverge from release-branch content and supported labels.  Worse, we won't learn about it breaking until somebody tries to open a (likely) critical CVE backport PR on a release branch :confounded: 

An alternative to restricting the trigger, could be to make this workflow behave more like Cirrus.  This can be accomplished by "backporting" this workflow file (and config YAML) to our important release-branches, under a different file name (and adding a `on: workflow_call`).  Then in this file (on `main`) [can "call" the wofkflow on the release branch](https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow) when a PR opens there, by using [the `${{github.head_ref}}` context.](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context).

Reply:
@cevich I'm sorry, I don't understand any of that. What I hear is you wanting to disable this except on `main`, is that correct? My concern with that is that I would expect it to be even **more** important to flag API changes in maintenance branches. But it sounds like your concern is a technical one. I don't need to understand it, since you're the github SME, so if you want to make the changes to run this only on `main`, please feel free.

Reply:
> I don't understand any of that.

Oof, well I tried my best to explain :(  GHA wasn't designed using POLA, and this is one of the nastier "gotchas".  Yes, it's more important on release-branches, and so the worst-place to suddenly discover the workflow has been broken for months.  Sure I can fix it, but I wanted to check and see if I was missing something or not.  Sounds like not.